### PR TITLE
[CI] Separated packaging step

### PR DESCRIPTION
**Pull request by pazone**
## What does this PR do?
 Separated packaging step

## Why is it important?
Now we don't need to re-package agent when restarting tests from BK UI

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Related issues
- Closes https://github.com/elastic/elastic-agent/issues/4142

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: .buildkite/scripts/steps/integration-package.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+source .buildkite/scripts/common.sh
+
+AGENT_PACKAGE_VERSION="$(cat .package-version)" PACKAGES=tar.gz,zip,rpm,deb PLATFORMS=linux/amd64,linux/arm64,windows/amd64  SNAPSHOT=true EXTERNAL=true  DEV=true mage package
```
Comment (pchila) [inline .buildkite/scripts/steps/integration-package.sh:6]: To take into account the changes in #5119 this line should be 
```suggestion
PACKAGES=tar.gz,zip,rpm,deb PLATFORMS=linux/amd64,linux/arm64,windows/amd64  SNAPSHOT=true EXTERNAL=true  DEV=true mage package
```

File: .buildkite/scripts/steps/integration_tests.sh [hunk 2]
```diff
@@ -8,14 +8,11 @@ MAGE_TARGET="${2:-"integration:test"}"
...
-
-STACK_VERSION="$(cat .package-version)"
-if [[ -n "$STACK_VERSION" ]]; then
-    STACK_VERSION=${STACK_VERSION}"-SNAPSHOT"
-fi
+# Override the agent package version using a string with format <major>.<minor>.<patch>
+# There is a time when the snapshot is not built yet, so we cannot use the latest version automatically
+# This file is managed by an automation (mage integration:UpdateAgentPackageVersion) that check if the snapshot is ready.
+OVERRIDE_AGENT_PACKAGE_VERSION="$(cat .package-version)"
+OVERRIDE_TEST_AGENT_VERSION=${OVERRIDE_AGENT_PACKAGE_VERSION}"-SNAPSHOT"
```
Comment (pchila) [inline .buildkite/scripts/steps/integration_tests.sh:15]:
```suggestion
# Override the stack version from `.package-version` contents
# There is a time when the current snapshot is not available on cloud yet, so we cannot use the latest version automatically
# This file is managed by an automation (mage integration:UpdateAgentPackageVersion) that check if the snapshot is ready.

STACK_VERSION="$(cat .package-version)"
if [[ -n "$STACK_VERSION" ]]; then
    STACK_VERSION=${STACK_VERSION}"-SNAPSHOT"
fi
```

**Review by pchila — COMMENTED** [review]
There are a couple of modifications needed to take #5119  into account

**Review by pchila — APPROVED** [review]
LGTM

**Review by michalpristas — APPROVED** [review]
looks good

**Pull request merged**